### ISSUE TITLE
Update via Packal

### ISSUE THREAD

**Issue by fractaledmind** [deanishe/alfred-workflow#31]
So, I've looked over @fniephaus' `update` addition to Alfred-Workflow, and I like it _a lot_. For me, however, it would add just another additional step to my workflow publishing. Now, I would need to 
1. upload on Packal
2. release on GitHub
3. post on Alfred

Now, I do try to keep my GitHub repos for my workflows up-to-date, but I'm not always on top of that. Plus, I don't currently "release" new versions via GitHub. That's what I use Packal for. So, I would love to have this functionality work with Packal as a backend, not the workflow's GitHub releases.

I am working on code to do just this; however, I am uncertain how best to integrate it into Alfred-Workflow. Obviously, I think but auto-update backends should be supported, but what's the simplest way to accomidate both? I was thinking, introduce classes to `update.py`. Have a `GitHub` class with @fniephaus' code and a `Packal` class with my code. Then add another argument to `update_info` when initializing a workflow, maybe `backend`? We would then also have to abstract away from the `github_slug` arg, since the Packal code requires the workflow's bundle id. But since `Alfred-Workflow` can get that automatically, maybe make `github_slug` only required if `backend=github` is specified. 

Basically, I want some community input on how to create a robust auto-updating feature that allows for multiple backends. You can see my initial mirror of @fniephaus' `update.py` in this [gist](https://gist.github.com/smargh/859ec011ca2ad652bacb).

Thoughts?
